builtin/logical/pki/path_config_ca.go

```diff
@@ -13,24 +16,443 @@ func pathConfigCA(b *backend) *framework.Path {
 		Pattern: "config/ca",
 		Fields: map[string]*framework.FieldSchema{
 			"pem_bundle": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: `DEPRECATED: use "config/ca/set" instead.`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.WriteOperation: b.pathCASetWrite,
+		},
+
+		HelpSynopsis:    pathConfigCASetHelpSyn,
+		HelpDescription: pathConfigCASetHelpDesc,
+	}
+}
+
+func pathSetCA(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/ca/set",
+		Fields: map[string]*framework.FieldSchema{
+			"pem_bundle": &framework.FieldSchema{
 				Type: framework.TypeString,
-				Description: `PEM-format, concatenated unencrypted secret key
-and certificate`,
+				Description: `PEM-format, concatenated unencrypted
+secret key and certificate, or, if a
+CSR was generated with the "generate"
+endpoint, just the signed certificate.`,
 			},
 		},
 
 		Callbacks: map[logical.Operation]framework.OperationFunc{
-			logical.WriteOperation: b.pathCAWrite,
+			logical.WriteOperation: b.pathCASetWrite,
+		},
+
+		HelpSynopsis:    pathConfigCASetHelpSyn,
+		HelpDescription: pathConfigCASetHelpDesc,
+	}
+}
+
+func pathGenerateRootCA(b *backend) *framework.Path {
+	ret := &framework.Path{
+		Pattern: "config/ca/generate/root/" + framework.GenericNameRegex("exported"),
+
+		Fields: rootAndSignSchema,
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.WriteOperation: b.pathCAGenerateRoot,
+		},
+
+		HelpSynopsis:    pathConfigCAGenerateHelpSyn,
+		HelpDescription: pathConfigCAGenerateHelpDesc,
+	}
+
+	for k, v := range generateSchema {
```

Comment:
Similarly, not sure what is happening here. Does the Fields need to be a copy instead of modified in place?